src/core/search/indices.cc

```diff
@@ -621,34 +652,43 @@ struct HnswlibAdapter {
 
   SpaceUnion space_;
   hnswlib::HierarchicalNSW<float> world_;
+  absl::Mutex resize_mutex_;
```

Comment:
why `resize_mutex_` is needed? I thought `HierarchicalNSW` is thread-safe

Reply:
but it is safe with reads/mark deletes? 🤔